libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway.hs

```diff
@@ -0,0 +1,875 @@
+
+instance SpecTranslate ctx a => SpecTranslate ctx (Maybe a) where
+  type SpecRep (Maybe a) = Maybe (SpecRep a)
+  type TestRep (Maybe a) = Maybe (TestRep a)
+
+  toSpecRep = traverse toSpecRep
+  specToTestRep = fmap (specToTestRep @ctx @a)
+
+instance SpecTranslate ctx (AlonzoTxAuxData era) where
+  type SpecRep (AlonzoTxAuxData era) = Agda.AuxiliaryData
+
+  toSpecRep _ = pure ()
+
+deriving instance SpecTranslate ctx (ScriptHash c)
+
+instance SpecTranslate ctx (Credential k c) where
+  type SpecRep (Credential k c) = Agda.Credential
+
+  toSpecRep (KeyHashObj h) = Agda.KeyHashObj <$> toSpecRep h
+  toSpecRep (ScriptHashObj h) = Agda.ScriptObj <$> toSpecRep h
+
+instance SpecTranslate ctx Network where
+  type SpecRep Network = ()
+
+  toSpecRep = pure . const ()
+
+instance SpecTranslate ctx (RewardAccount c) where
+  type SpecRep (RewardAccount c) = Agda.RwdAddr
+
+  toSpecRep (RewardAccount n c) = toSpecRep (n, c)
+
+instance SpecTranslate ctx (PoolParams era) where
+  type SpecRep (PoolParams era) = Agda.PoolParams
+
+  toSpecRep PoolParams {..} = toSpecRep $ KeyHashObj ppId
+
+instance SpecTranslate ctx (DRep c) where
+  type SpecRep (DRep c) = Agda.VDeleg
+
+  toSpecRep (DRepCredential c) = Agda.CredVoter Agda.DRep <$> toSpecRep c
+  toSpecRep DRepAlwaysAbstain = pure Agda.AbstainRep
+  toSpecRep DRepAlwaysNoConfidence = pure Agda.NoConfidenceRep
+
+instance SpecTranslate ctx (Anchor c) where
+  type SpecRep (Anchor c) = Agda.Anchor
+  toSpecRep _ = pure ()
+
+instance SpecTranslate ctx (ConwayTxCert era) where
+  type SpecRep (ConwayTxCert era) = Agda.TxCert
+
+  toSpecRep (ConwayTxCertDeleg (ConwayRegCert _ _)) = throwError "RegCert not supported"
+  toSpecRep (ConwayTxCertDeleg (ConwayUnRegCert _ _)) = throwError "UnRegCert not supported"
+  toSpecRep (ConwayTxCertDeleg (ConwayDelegCert c d)) =
+    Agda.Delegate
+      <$> toSpecRep c
+      <*> toSpecRep (getVoteDelegatee d)
+      <*> toSpecRep (KeyHashObj <$> getStakePoolDelegatee d)
+      <*> pure 0
+  toSpecRep (ConwayTxCertDeleg (ConwayRegDelegCert s d c)) =
+    Agda.Delegate
+      <$> toSpecRep s
+      <*> toSpecRep (getVoteDelegatee d)
+      <*> toSpecRep (KeyHashObj <$> getStakePoolDelegatee d)
+      <*> toSpecRep c
+  toSpecRep (ConwayTxCertPool (RegPool p@PoolParams {ppId})) =
+    Agda.RegPool
+      <$> toSpecRep (KeyHashObj ppId)
+      <*> toSpecRep p
+  toSpecRep (ConwayTxCertPool (RetirePool kh e)) =
+    Agda.RetirePool
+      <$> toSpecRep (KeyHashObj kh)
+      <*> toSpecRep e
+  toSpecRep (ConwayTxCertGov (ConwayRegDRep c d _)) =
+    Agda.RegDRep
+      <$> toSpecRep c
+      <*> toSpecRep d
+      <*> pure () -- TODO Are the anchors supposed to be optional?
+  toSpecRep (ConwayTxCertGov (ConwayUnRegDRep c _)) =
+    Agda.DeRegDRep
+      <$> toSpecRep c
+  toSpecRep (ConwayTxCertGov (ConwayUpdateDRep c _)) =
+    Agda.RegDRep
+      <$> toSpecRep c
+      <*> pure 0
+      <*> pure () -- TODO Are the anchors supposed to be optional?
```

Comment:
Yeah, that's fair, I was asking this because we use a `StrictMaybe` here in the implementation but not in the spec. I guess it doesn't really matter because the rules don't look at the anchors anyways.